va/validation-authority.go

```diff
@@ -398,108 +360,6 @@ func (va *ValidationAuthorityImpl) validateTLSWithZName(identifier core.AcmeIden
 	return challenge, challenge.Error
 }
 
-// TODO(https://github.com/letsencrypt/boulder/issues/894): Delete this method
-func (va *ValidationAuthorityImpl) validateSimpleHTTP(identifier core.AcmeIdentifier, input core.Challenge) (core.Challenge, error) {
-	challenge := input
-
-	if identifier.Type != core.IdentifierDNS {
-		challenge.Status = core.StatusInvalid
-		challenge.Error = &core.ProblemDetails{
-			Type:   core.MalformedProblem,
-			Detail: "Identifier type for SimpleHTTP was not DNS",
-		}
-
-		va.log.Debug(fmt.Sprintf("SimpleHTTP [%s] Identifier failure", identifier))
-		return challenge, challenge.Error
-	}
-
-	// Perform the fetch
-	path := fmt.Sprintf(".well-known/acme-challenge/%s", challenge.Token)
-	useTLS := (challenge.TLS == nil) || *challenge.TLS
-	body, challenge, err := va.fetchHTTP(identifier, path, useTLS, challenge)
-	if err != nil {
-		return challenge, err
-	}
-
-	// Parse and verify JWS
-	parsedJws, err := jose.ParseSigned(string(body))
-	if err != nil {
-		err = fmt.Errorf("Validation response failed to parse as JWS: %s", err.Error())
-		va.log.Debug(err.Error())
-		challenge.Status = core.StatusInvalid
-		challenge.Error = &core.ProblemDetails{
-			Type:   core.UnauthorizedProblem,
-			Detail: err.Error(),
-		}
-		return challenge, err
-	}
-
-	// Check that JWS body is as expected
-	// * "type" == "simpleHttp"
-	// * "token" == challenge.token
-	// * "tls" == challenge.tls || true
-	target := map[string]interface{}{
-		"type":  core.ChallengeTypeSimpleHTTP,
-		"token": challenge.Token,
-		"tls":   (challenge.TLS == nil) || *challenge.TLS,
-	}
-	err = verifyValidationJWS(parsedJws, challenge.AccountKey, target)
-	if err != nil {
-		va.log.Debug(err.Error())
-		challenge.Status = core.StatusInvalid
-		challenge.Error = &core.ProblemDetails{
-			Type:   core.UnauthorizedProblem,
-			Detail: err.Error(),
-		}
-		return challenge, err
-	}
-
-	challenge.Status = core.StatusValid
-	return challenge, nil
-}
-
-// TODO(https://github.com/letsencrypt/boulder/issues/894): Delete this method
-func (va *ValidationAuthorityImpl) validateDvsni(identifier core.AcmeIdentifier, input core.Challenge) (core.Challenge, error) {
```

Comment:
Fair point.  I guess we can split this in two, but given that pending authz are only valid for a week, we should be able to land the second half shortly after the first.